KarhooSDK/Service/Loyalty/LoyaltyPreAuth/KarhooLoyaltyPreAuthInteractor.swift

```diff
@@ -0,0 +1,48 @@
+//
+//  KarhooLoyaltyPreAuthInteractor.swift
+//  KarhooSDK
+//
+//  Created by Edward Wilkins on 18/11/2021.
+//  Copyright © 2021 Flit Technologies Ltd. All rights reserved.
+//
+
+import Foundation
+
+final class KarhooLoyaltyPreAuthInteractor: LoyaltyPreAuthInteractor {
+    
+    private let requestSender: RequestSender
+    private var identifier: String?
+    private var preAuthPayload: LoyaltyPreAuthPayload?
+    
+    init(requestSender: RequestSender = KarhooRequestSender(httpClient: TokenRefreshingHttpClient.shared)) {
+        self.requestSender = requestSender
+    }
+    
+    func set(loyaltyPreAuth: LoyaltyPreAuthPayload) {
+        self.preAuthPayload = loyaltyPreAuth
+    }
+    
+    func execute<T>(callback: @escaping CallbackClosure<T>) where T : KarhooCodableModel {
+        guard let preAuthPayload = self.preAuthPayload else {
+            return
+        }
+        
+        let payload = LoyaltyPreAuthPayload(currency: preAuthPayload.currency, points: preAuthPayload.points, flexpay: preAuthPayload.flexpay, membership: preAuthPayload.membership)
+        
+        requestSender.request(payload: payload,
+                              endpoint: endpoint(identifier: identifier ?? ""),
```

Comment:
At which point do you set the value of this identifier? My proposal to fix this is: 
1. Create a request object (LoyaltyPreAuth) that contains the identifier, currency, points, flexpay and membership (basically combine all the information needed for this API call). 
2. Set it using the method at line 21 (change the parameter to take the request object instead of the payload as a parameter). 
3. Store the object instead of the identifier and payload at lines 14 and 15. 
4. Then, on line 30, create the payload object using the request object and on line 33 pass the identifier from the request object. 
I've used KarhooCancelTripInteractor as a pattern since the API structure is the same as what we have here, i.e.: POST with a path parameter (id) and a body parameter (payload).

Side note: At the moment on line 30 you're creating a new object identical to the payload you already have at that point. That seems redundant, but will change with what I proposed above.